### PR TITLE
Ticket #6771 - add order field to task and allow user to order tasks

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

File: app/graph/mutations/tasks_order_mutation.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+TasksOrderMutation = GraphQL::Relay::Mutation.define do
```
Comment (infojunkie) [inline app/graph/mutations/tasks_order_mutation.rb:1]: Why this and not the usual `set_fields` to set the order via the regular `UpdateTaskMutation` ? @caiosba

File: app/models/annotations/task.rb [hunk 2]
```diff
@@ -28,6 +28,7 @@ def self.task_types
   field :suggestions_count, Integer
   field :pending_suggestions_count, Integer
   field :team_task_id, Integer
+  field :order, Integer
```
Comment (infojunkie) [inline app/models/annotations/task.rb:31]: where is this field stored?

File: app/graph/mutations/tasks_order_mutation.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  input_field :tasks, !JsonStringType
+
+  return_field :success, types.Boolean
+
+  return_field :errors, JsonStringType
+
+  resolve -> (_root, inputs, _ctx) {
+    errors = Task.order_tasks(inputs[:tasks])
+    { success: true, errors: errors }
```
Comment (infojunkie) [inline app/graph/mutations/tasks_order_mutation.rb:12]: `success: !errors`

File: app/graph/mutations/tasks_order_mutation.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+TasksOrderMutation = GraphQL::Relay::Mutation.define do
+  name 'TasksOrder'
+
+  input_field :tasks, !JsonStringType
```
Comment (infojunkie) [inline app/graph/mutations/tasks_order_mutation.rb:4]: how is the order specified in this mutation?

File: app/graph/mutations/tasks_order_mutation.rb (hunk 1 above)
Comment (infojunkie) [reply app/graph/mutations/tasks_order_mutation.rb:1]: oh, I see this was specified in the ticket - sorry

**Comment by infojunkie** [comment]
I think a big piece missing here - and sorry if it was missing from the ticket - is to return the tasks ordered according to `order` in the `ProjectMediaType.tasks` call

File: app/models/annotations/task.rb (hunk 2 above)
Comment (melsawy) [reply app/models/annotations/task.rb:31]: In Data column

File: app/graph/mutations/tasks_order_mutation.rb (hunk 3 above)
Comment (melsawy) [reply app/graph/mutations/tasks_order_mutation.rb:12]: I did a bulk sort so may be one of the ids have an issue like record not found, user has no ability to update, etc ... so I catch any errors while saving the item in errors key

**Pull request merged**